src/Microsoft.SqlTools.ResourceProvider.DefaultImpl/AzureAuthenticationManager.cs

```diff
@@ -0,0 +1,314 @@
+            get { return false; }
+        }
+
+        /// <summary>
+        /// Set current logged in user
+        /// </summary>
+        public async Task<IUserAccount> SetCurrentAccountAsync(object account)
+        {
+            CommonUtil.CheckForNull(account, nameof(account));
+            AccountTokenWrapper accountTokenWrapper = account as AccountTokenWrapper;
+            if (accountTokenWrapper != null)
+            {
+                AzureUserAccount userAccount = CreateUserAccount(accountTokenWrapper);
+                accountsMap[userAccount.UniqueId] = userAccount;
+                currentAccountId = userAccount.UniqueId;
+            }
+            else
+            {
+                throw new ServiceFailedException(string.Format(CultureInfo.CurrentCulture, SR.UnsupportedAuthType, account.GetType().Name));
+            }
+            OnCurrentAccountChanged();
+            return await GetCurrentAccountAsync();
+        }
+        
+        /// <summary>
+        /// Public for testing purposes. Creates an Azure account with the correct set of mappings for tenants etc.
+        /// </summary>
+        /// <param name="accountTokenWrapper"></param>
+        /// <returns></returns>
+        public AzureUserAccount CreateUserAccount(AccountTokenWrapper accountTokenWrapper)
+        {
+            Account account = accountTokenWrapper.Account;
+            CommonUtil.CheckForNull(accountTokenWrapper.Account, nameof(account));
+            CommonUtil.CheckForNull(accountTokenWrapper.SecurityTokenMappings, nameof(account) + ".SecurityTokenMappings");
+            AzureUserAccount userAccount = new AzureUserAccount();
+            userAccount.UniqueId = account.Key.AccountId;
+            userAccount.DisplayInfo = ToDisplayInfo(account);
+            IList<IAzureTenant> tenants = new List<IAzureTenant>();
+            foreach (Tenant tenant in account.Properties.Tenants)
+            {
+                string token;
+                if (accountTokenWrapper.SecurityTokenMappings.TryGetValue(tenant.Id, out token))
+                {
+                    AzureTenant azureTenant = new AzureTenant()
+                    {
+                        TenantId = tenant.Id,
+                        AccountDisplayableId = tenant.DisplayName,
+                        AccessToken = token
+                    };
+                    tenants.Add(azureTenant);
+                }
+                // else ignore for now as we can't handle a request to get a tenant without an access key
+            }
+            userAccount.AllTenants = tenants;
+            return userAccount;
+        }
+
+        private AzureUserAccountDisplayInfo ToDisplayInfo(Account account)
+        {
+            return new AzureUserAccountDisplayInfo()
+            {
+                AccountDisplayName = account.DisplayInfo.DisplayName,
+                ProviderDisplayName = account.Key.ProviderId
+            };
+        }
+
+        private void OnCurrentAccountChanged()
```

Comment:
Overall, I'm still not quite sure how this "current account" stuff is intended to work. It's probably a higher level architecture question, so I won't have this block to review. But, we should think about it a bit more, since I have a suspicion that it's a holdover from SSDT implementation.

Reply:
This is all ported over. There's basically 1 active account for most Azure operations and this is an intended behavior. While you can have knowledge of a lot of accounts, you choose one (via the UI) that is the current primary and then all operations are simple.